### PR TITLE
Adding clockwork to recurrently fetch feeds

### LINKED ISSUE #411: Improve robustness of rake work_jobs
First of all, many thanks on creating this awesome RSS reader! Just what I was looking for and was about to create myself, but Stringer is so much better than anything I wanted :)

I installed it into Heroku and noticed a problem with reliability of the Heroku-specific `rake lazy_fetch` approach:

While the `bundle exec rake work_jobs` is launched from config/unicorn.rb, it's never checked to be running and its delayed_job_pid is never used. It seems that Heroku kills long-running database connections either on purpose, or just randomly, which also causes an unhandled exception to kill the `bundle exec rake work_jobs` process ([example log](https://github.com/swanson/stringer/files/123722/ConnectionLoss.txt)). While `rake lazy_fetch` continues to submit jobs, the worker process is dead and never restarted.

It seems the best way would be to run `rake work_jobs` [from a Procfile](http://stackoverflow.com/a/14855937/2401011). I've tried two other workarounds and can confirm they work:

1. Restart the rake process in bash:
```
diff --git config/unicorn.rb config/unicorn.rb
--- config/unicorn.rb
+++ config/unicorn.rb
@@ -9,7 +9,7 @@ before_fork do |_server, _worker|
   # as there's no need for the master process to hold a connection
   ActiveRecord::Base.connection.disconnect! if defined?(ActiveRecord::Base)
 
-  @delayed_job_pid ||= spawn("bundle exec rake work_jobs")
+  @delayed_job_pid ||= spawn("while true; do bundle exec rake work_jobs; sleep 60; done")
 
   sleep 1
 end
```
2. Catch exceptions in the work_jobs job and try to reconnect:
```
diff --git Rakefile Rakefile
--- Rakefile
+++ Rakefile
@@ -46,11 +46,25 @@ desc "Work the delayed_job queue."
 task :work_jobs do
   Delayed::Job.delete_all
 
-  3.times do
-    Delayed::Worker.new(
-      min_priority: ENV["MIN_PRIORITY"],
-      max_priority: ENV["MAX_PRIORITY"]
-    ).start
+  logger = Logger.new(STDOUT)
+  logger.level = Logger::DEBUG
+
+  loop do
+    begin
+      logger.info "(Re-)starting worker"
+      if !ActiveRecord::Base.connection.active?
+        logger.info "Restarting DB connection"
+        ActiveRecord::Base.connection.reconnect!
+      end
+      Delayed::Worker.new(
+        min_priority: ENV["MIN_PRIORITY"],
+        max_priority: ENV["MAX_PRIORITY"]
+      ).start
+    rescue => e
+      logger.error e.message
+      e.backtrace.each { |line| logger.error line }
+      sleep 60
+    end
   end
 end
```

### PR DESCRIPTION
**Pull request by fearenales**
This PR intends to add [`clockwork` gem](https://rubygems.org/gems/clockwork/versions/1.2.0) to recurrently fetch feeds.

It will replace [Scheduler](https://elements.heroku.com/addons/scheduler) (for deployment on Heroku), so that no extra step is needed due to another `spawn` command being called on `unicorn.rb` starting `clockwork`.

Still on Heroku, `Procfile` is updated, describing the `worker` and `clock` components (if the user wants a more robust architecture). This fixes #411 

Also, for development environments, it will allow fetching feeds without manually running `rake` commands.

The default interval for fetching feeds is 10 min and it can be customized by env var `FETCH_INTERVAL`.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,6 +27,7 @@ end
 gem "activerecord", "~> 4.1.11"
 gem "arel", "~> 5.0"
 gem "bcrypt-ruby", "~> 3.1.2"
+gem "clockwork", "~> 1.2.0"
```
Comment (Koronen) [inline Gemfile]: Can we lock this gem at the major version, see c3e41b4d1476f0d7bf9a45084ff5d979269b2ae9?

File: db/schema.rb [hunk 2]
```diff
@@ -25,8 +25,8 @@
     t.datetime "failed_at"
     t.string   "locked_by"
     t.string   "queue"
-    t.datetime "created_at",             null: false
-    t.datetime "updated_at",             null: false
+    t.datetime "created_at"
+    t.datetime "updated_at"
```
Comment (Koronen) [inline db/schema.rb]: Revert these seemingly unrelated changes to `db/schema.rb`?

File: lib/clock.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+require "delayed_job"
+require "delayed_job_active_record"
+
+require_relative "../app"
+require_relative "../app/jobs/fetch_feed_job"
+require_relative "../app/repositories/feed_repository"
+
+include Clockwork
+
+fetch_interval = (ENV["FETCH_INTERVAL"] || 60).to_i
```
Comment (Koronen) [inline lib/clock.rb]: Fetching new stories every minute might be too often? What do we think about increasing the default interval to something like once every 10/30/60 minutes (as stated in `docs/VPS.md`/-/`docs/OpenShift.md` respectively)?

**Comment by Koronen** [comment]
I can see this being a useful changeset. Thank you for your contribution! I'll give this a spin on my instance.

**Comment by Koronen** [comment]
Hmm... One issue with these changes is that for the app to be fully functional on Heroku without using the embedded workers, you would have to run three dynos. Heroku only allows for two concurrent Free-size dynos. Any thoughts on this?

File: Gemfile (hunk 1 above)
Comment (fearenales) [inline Gemfile]: Sure! :)

File: db/schema.rb (hunk 2 above)
Comment (fearenales) [inline db/schema.rb]: Done.

File: lib/clock.rb (hunk 3 above)
Comment (fearenales) [inline lib/clock.rb]: OK, I've used the VPS strategy and used 10 min as default fetching frequency. Does it sounds good?

**Comment by fearenales** [comment]
@Koronen Why not using embedded workers on Heroku? That should work.
And if you need a more robust infrastructure, just deploy 3 of them.

**Comment by swanson** [comment]
I'm not sure what adding `clockwork` gets us -- Heroku (the preferred deployment platform) offers a robust scheduler already and any other Unix-y platform will have a cronjob service. I don't like taking dependencies in the project if we can avoid it.

**Comment by gabrielpoca** [comment]
I'm closing this pull request for lack of activity.
